### PR TITLE
Update the reload pass to replace copies with fill/spill instructions

### LINKED ISSUE #245: Reload: Elide copy instructions
Copy instructions are inserted by two regalloc passes before the reload pass:

1. The coalescer can insert copies before branches to separate virtual registers that would conflict.
2. The spiller can insert copies before two-address instructions, and to deal with other tricky constraints.

The reload pass should handle these copy instructions better when one or both values have been spilled.

```
v10 = copy v5
```

Currently becomes:

```
v50 = fill v5
v10 = copy v50
```

or

```
v50 = copy v5
v10 = spill v50
```

These copies are not necessary. If the reload pass recognized `copy` instructions, we could do better:

```
v10 = fill v5
```

or

```
v10 = spill v5
```

In the case where both values were spilled, we need a stack-stack copy:

```
v50 = fill v5
v10 = spill v50
```

### PR DESCRIPTION
**Pull request by mlea13**
This fixes #245.

### CODE REVIEW

File: filetests/regalloc/reload.clif [hunk 1]
```diff
@@ -17,3 +17,30 @@ ebb0:
...
+function %spilled_copy_result(i32) -> i32 {
+    fn0 = %foo(i32)
+
+ebb0(v0: i32):
+    ; not: copy
+    ; check: v1 = spill v0
+    v1 = copy v0
+    call fn0(v1)
+    return v1
+}
```
Comment (bjorn3) [inline filetests/regalloc/reload.clif]: Missing trailing newline

File: lib/codegen/src/regalloc/reload.rs [hunk 2]
```diff
@@ -209,40 +209,14 @@ impl<'a> Context<'a> {
...
-            let reg = self.cur.ins().fill(cand.value);
-            let fill = self.cur.built_inst();
-
-            self.reloads.insert(ReloadedValue {
-                stack: cand.value,
-                reg,
-            });
-            cand.value = reg;
-
-            // Create a live range for the new reload.
```
Comment (sunfishcode) [inline lib/codegen/src/regalloc/reload.rs]: I'm curious why you removed this comment.

Comment (mlea13) [reply lib/codegen/src/regalloc/reload.rs]: Oops, meant to copy that over as well.

File: filetests/regalloc/reload.clif (hunk 1 above)
Comment (mlea13) [reply filetests/regalloc/reload.clif]: Fixed.

**Comment by sunfishcode** [comment]
Thanks for working on this! A few tests I tried this on saw reductions of the number of instructions of 9%.

**Pull request merged**